### ISSUE TITLE
How to cast created entity by PooledEngine to another class extends Entity

### ISSUE THREAD

**Issue by ronjunelopez** [libgdx/ashley#234]
Currently I am using  [builder-pattern](https://en.wikipedia.org/wiki/Builder_pattern) to create my `Entities`

    PooledEngine engine = new PooledEngine();
    
    ...

    CharacterComponent character = engine.createComponent(CharacterComponent.class);
    TextureComponent texture = engine.createComponent(TextureComponent.class);
    SizeComponent size = engine.createComponent(SizeComponent.class);
    TransformComponent transform = engine.createComponent(TransformComponent.class);
    PhysicsComponent physics = engine.createComponent(PhysicsComponent.class);

    Entity entity = engine.createEntity();
    entity.flag = Constants.CHARACTER;

    entity.add(character);
    entity.add(texture);
    entity.add(size);
    entity.add(transform);
    entity.add(physics);

    engine.addEntity(entity);

Because `PooledEngine` class didnt support the creation of casting an Entity to another class that extends Entity. Example `engine.createEntity(Character.class);`, I am having hard thinking how do I wrap these components into single class.

**Comment by saltares** [comment]
Why do you need to wrap a component configuration as a class that extends `Entity`? That doesn't make a lot of sense to me... You'd be losing the advantages of ECS.

**Comment by ronjunelopez** [comment]
I'm not sure if its a good idea, because I'm used to OOP concept that's why I wanted to wrap its component to a class that extends `Entity`. Because I am having hard time syncing the `Child Entity` to `Parent Entity`. My `character` `component`  has a field called `Entity gun`, I want this entity to become `Gun gun` then inside the `Gun` there is a method reload, fire, trigger etc.

**Comment by saltares** [comment]
You're mixing the OOP approach with the ECS approach. In ECS you have components that only hold data and systems that operate on components adding the logic. Entity should just be a collection of components, nothing more.

**Comment by ronjunelopez** [comment]
As the result of a few research, I end up using builder-pattern. I guess it's much better than wrapping the entity to class. As you were saying I just mixed OOP and ECS, you're right I'm missing its advantage! :+1: . I've also ask the same question in [stackoverflow](www.stackoveerflow.com) and the result is same. I finally decided what to use ([source](http://gamedev.stackexchange.com/questions/125651/how-to-minimize-the-repetitive-creation-of-component-in-libgdx-ashley/125662#125662)). Thanks
